### ISSUE TITLE
[bug] cmake helper is handling Visual Studio 16 2019 incorrectly

### ISSUE THREAD

**Issue by Carsten87** [conan-io/conan#7683]
With CMake generator "Visual Studio 16 2019" the specification of the architecture changed. It must now be supplied via the -A option. It seems like conan is not handling this correctly - note the `Win64` which is appended (probably by conan based on the specified settings):

>CMake Error: Could not create named generator Visual Studio 16 2019 **Win64**
>
>Generators
>* Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
>                                Use -A option to specify architecture.
> Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
>                                 Optional [arch] can be "Win64" or "ARM".

My settings are:
>[settings]
>arch=x86_64
>arch_build=x86_64
>build_type=Debug
>compiler=Visual Studio
>compiler.runtime=MDd
>compiler.toolset=v141
>compiler.version=15
>os=Windows
>os_build=Windows

My `build` method looks as follows:

~~~python
    def build(self):
        cmake = CMake(self, generator="Visual Studio 16 2019")
        cmake.configure()
        cmake.build()
~~~

I tried it with conan version 1.28.1 and 1.29.0. Generator "Visual Studio 15 2017" is working fine.

**Comment by memsharded** [comment]
Hi @Carsten87 

Thanks for submitting this report.
What is exactly the error that you are getting? I am reproducing with a test, and what I see is that the produced command line is ``'-G "Visual Studio 16 2019" -A "x64"`` instead of the ``-A "Win64"``. Is that the case?

UPDATE: Sorry, I skipped the ``CMake Error: Could not create named generator Visual Studio 16 2019 Win64`` error. I don't see yet how this is happening, will keep having a look.

**Comment by Carsten87** [comment]
Hi @memsharded 

thank you for looking into it. I will also try to reproduce it again and maybe give more detailed instructions.

**Comment by memsharded** [comment]
I have been able to reproduce, and submitted a possible fix in https://github.com/conan-io/conan/pull/7684.
It might be possible that this cannot be merged, it could be breaking, we need to make sure, but lets see how it goes.

**Comment by Carsten87** [comment]
@memsharded wow, you are quick. Thank you 👍 

Workaround is to just set `compiler.version=16` in the settings.

**Comment by memsharded** [comment]
Hi @Carsten87 

this has been fixed and released in 1.29.1. Please update, give it a try, and let us know if all is ok.

**Comment by Carsten87** [comment]
@memsharded thank you. I can confirm that it is fixed. I tried with 1.29.2
